File: python/Day20_Sorting.py
#!/bin/python3

import sys

n = int(input().strip())
a = list(map(int, input().strip().split(' ')))
# Write Your Code Here

swap_count = 0

for i in range(len(a)-1, 0, -1):
    for j in range(0,i):
        if a[j] > a[j+1]:
            a[j], a[j+1] = a[j+1], a[j]
            swap_count += 1

print("Array is sorted in {} swaps.".format(swap_count))
print("First Element: {}".format(a[0]))
print("Last Element: {}".format(a[-1]))

File: scala/Day20_Sorting.scala
object Solution {

    def main(args: Array[String]) {
        val sc = new java.util.Scanner (System.in);
        var n = sc.nextInt();
        var a = new Array[Int](n);
        var swap_count = 0
        for(a_i <- 0 to n-1) {
        	a(a_i) = sc.nextInt();
        }
        // Write Your Code Here
        for(i <- a.size-1 to 0 by -1) {
            for(j <- 0 to i-1) {
                if(a(j) > a(j+1)){
                    val temp = a(j)
                    a(j) = a(j+1)
                    a(j+1) = temp
                    swap_count += 1
                }
            }
        }
        println("Array is sorted in " + swap_count +" swaps.")
        println("First Element: " + a(0))
        println("Last Element: " + a(n-1))
    }
}